File: 프로그래머스/lv1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    
    for i in commands:
        array2 = array[i[0]-1:i[1]]
        array2.sort()
        # 5,6번째 줄 합치면 값이 잘 안나옴
        # array2 = array[i[0]-1:i[1]].sort()
        
        answer.append(array2[i[2]-1])
    
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];

        for (int i = 0; i < commands.length; i++) {
            answer[i] = result(array, commands[i]);
        }

        return answer;
    }

    public int result(int[] array, int[] command) {
        int i = command[0];
        int j = command[1];
        int k = command[2];

        // 배열 복사
        int[] result = Arrays.copyOfRange(array, i - 1, j);

        // 배열 정렬
        Arrays.sort(result);

        return result[k - 1];
    }
}
